### ISSUE TITLE
Too many false positives 

### ISSUE THREAD

**Issue by cnandreu** [hortinstein/node-dash-button#32]
For what it's worth I realize this library is [only watching ARP requests](https://github.com/hortinstein/node-dash-button#node-dash-button) and anything more is probably [beyond the scope of this project](https://github.com/hortinstein/node-dash-button/issues/13).

I post this in hopes that someone has run across the same issue and can provide insight, maybe translating to some updates to the README for future readers.

I got an Amazon Dash button yesterday and went through the setup process described in the README. Basically I did all the steps with the exception of the "select a product" step and [turned off notifications on the app](https://davekz.com/hacking-amazon-dash-buttons/). Then I found my button's mac address with `node bin/findbutton` and wrote the following:

```
var dash_button = require("node-dash-button");
var dash = dash_button(DASH_BTN_MAC, "en0", 60000);

dash.on("detected", function (dash_id) {
    console.log(new Date().toString() + "Dash button was clicked!");
});
```

That worked great. When I clicked the button the light flashed and registered the console message.

I made sure to specify a timeout to "debounce" (1min) the call and made sure to specify the network interface just to be explicit.

I left the app running for a couple of hours and saw this in the logs:

```
$ sudo node app.js 
Sat Jun 18 2016 17:21:29 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 17:26:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 17:36:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 17:41:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 17:46:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 17:51:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 17:56:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 18:01:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 18:06:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 18:16:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 18:21:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 18:22:23 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 18:26:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 18:36:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 18:41:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 18:46:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 18:56:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 19:01:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 19:06:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 19:16:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 19:21:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 19:26:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 19:36:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 19:46:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 19:51:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:01:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:05:40 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:06:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:11:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:11:54 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:16:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:20:47 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:21:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:26:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:27:48 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:31:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:41:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:46:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:51:30 GMT-0500 (CDT)Dash button was clicked!
Sat Jun 18 2016 20:54:34 GMT-0500 (CDT)Dash button was clicked!
```

**None** of those events were cause by manually clicking the button, it looks like it's trying to reach out to the server every 5 minutes (?).

**I'm wondering if anyone has run into this or has any insight with regards to dealing with false positives?**

I'm using node v6.2.1, npm 3.9.3 and node-dash-button 0.5.0 on a mac mini running OS X 10.11.5.

**Comment by SoullessWaffle** [comment]
My guess is that the `findbutton` script selected the wrong mac address, and when you pressed the button the device with that mac address also happened to ARP. I have two Dash buttons set up and neither one is giving any false positives.

**Comment by hortinstein** [comment]
yeah, see #36 ...I think this might help people.  I may make the findbutton script a little bit more robust

**Comment by nobetw** [comment]
Side tool - use your router admin panel to narrow-down the device address. Then you can guarantee the results.

**Comment by hortinstein** [comment]
closed due to #35 merge
